manageiq-operator/api/v1alpha1/helpers/miq-components/httpd.go

```diff
@@ -414,6 +415,8 @@ func HttpdDeployment(client client.Client, cr *miqv1alpha1.ManageIQ, scheme *run
 		"app":  cr.Spec.AppName,
 		"name": "httpd",
 	}
+	deploymentSelectorLabels := map[string]string{}
+	maps.Copy(deploymentLabels, deploymentSelectorLabels)
```

Comment:
Is this only on the initial deployment? A function name like `NewMemcachedDeployment` imply only the initial one but wasn't sure if the copying is ever happening after the initial deployment creation.

Reply:
So the operator sdk is weird. This method sets up a bunch of objects and then it returns a mutation function that is run every time a reconcile needs to happen. As such, this code outside of the function is in a closure bound to that function. For all intents and purposes it's like a global object and a function that manipulates the global object. It's super weird.